astropy/table/operations.py

```diff
@@ -61,6 +61,8 @@ def _get_list_of_tables(tables):
 
     # Convert inputs (Table, Row, or anything column-like) to Tables.
     # Special case that Quantity converts to a QTable.
+    # Do this in a separate list to not modify the original input list
+    tables = list(tables)
```

Comment:
This is a nice catch! I'm also fine with it, but I think the code would be clearer if we just constructed a new list as we went, like
```
result = []
for val in tables:
    if isinstance(val, Table):
        pass  # short-circuit common case
    elif isinstance(val, Row):
        val = Table(val)
    elif ...
    ...
    result.append(val)

return result
```
What do you think?

Reply:
It's admittedly a matter of taste.  While I agree that your suggestion is correct and guarantees that the inputs aren't modified, I also tend to be suspicious of code that assembles lists with `list.append` in `if/elif/then` chains because it risks future bugs with a mis-placed `continue` statement or some logic that otherwise accidentally misses the `append`.

But I will defer to you as an astropy maintainer to make the style call, and update the code if you feel strongly enough about it (I don't feel strongly about it).  You should have write access to the source branch.